### ISSUE TITLE
loss non-finite during SFT training

### ISSUE THREAD

**Issue by wolf943134497** [AIGeeksGroup/3D-R1#40]
Hi @Believeht029 @steve-zeyu-zhang  Thanks for your great works!

during the SFT stage, I encountered the following issue in Epoch 0：

Loss in not finite. Skip this training step.
Loss in not finite. Skip this training step.
Loss in not finite. Skip this training step.
Loss in not finite. Skip this training step.
Loss in not finite. Terminate training.

My running command is as follows:

torchrun \
    --nproc_per_node=4 \
    --master_port=29500 \
    main.py \
    --checkpoint_dir ./checkpoints/fast_eval \
    --dataset scenecold_dataset \
    --vocab "qwen/Qwen2.5-7B" \
    --qformer_vocab "bert-base-uncased" \
    --captioner 3dr1 \
    --detector point_encoder \
    --max_des_len 256 \
    --max_gen_len 64 \
    --batchsize_per_gpu 4 \
    --base_lr 1e-4 \
    --final_lr 1e-6 \
    --weight_decay 0.01 \
    --clip_gradient 1.0 \
    --warm_lr 1e-7 \
    --warm_lr_epochs 10 \
    --max_epoch 50 \
    --log_every 10 \
    --save_every 1000 \
    --eval_every_iteration 2000 \
    --start_eval_after 1000 \
    --criterion "CiDEr" \
    --use_color \
    --use_multiview \
    --use_additional_encoders \
    --use_depth \
    --use_image \
    --max_multiview_images 4 \
    --max_multiview_depth 4 \
    --depth_encoder_dim 256 \
    --image_encoder_dim 256 \
    --use_lora \
    --lora_r 16 \
    --lora_alpha 32 \
    --lora_dropout 0.1 \
    --enable_dynamic_views \
    --view_selection_weight 0.05 \
    --eval_batch_size 8 \
    --eval_max_samples 1000 \
    --eval_use_fp16 \
    --eval_skip_metrics \
    --seed 42

Could you please provide any suggestions for resolving this issue? Thanks a lot！

Another issue is how to use the Qwen2.5-VL-7B model—I configured the code to use_multimodal_model parameter, but it throws an error; the code seems unable to load the vision and language parameters, the model should be loaded using Qwen2VLForConditionalGeneration and Qwen2VLProcessor, right?

**Comment by Believeht029** [comment]
Thanks for the detailed report and for sharing your launch command.

- First, double-check the data loading pipeline: make sure the LiDAR/image/depth tensors don’t contain NaNs or Infs before they reach the model (add quick torch.isfinite() assertions right after they’re loaded).
- NaNs can also arrive from the loss itself. Try running one batch with torch.autograd.set_detect_anomaly(True) to catch the exact op that blows up.
- Gradient overflow is another common culprit. Since you’re already clipping, try temporarily disabling mixed precision (if it’s on) or switch the training loss to FP32 for a few steps to see if the problem disappears. Reducing --base_lr to something like 5e-5 and shrinking the warm-up range (e.g., fewer warmup epochs or a slightly higher --warm_lr) can also help stabilize the first epoch.
- If you’re mixing many optional views (--use_multiview, --use_depth, etc.), try turning them off and adding them back one by one; that often flushes out which branch is injecting bad values.
